packages/botonic-plugin-contentful/src/contentful/manage/manage-environment.ts

```diff
@@ -0,0 +1,72 @@
+import { createClient } from 'contentful-management'
+import { ClientAPI } from 'contentful-management/dist/typings/create-contentful-api'
+import {
+  Collection,
+  Environment,
+  EnvironmentAlias,
+  EnvironmentProps,
+  Space,
+} from 'contentful-management/dist/typings/export-types'
+
+import { ContentfulOptions } from '../../plugin'
+
+export class ManageEnvironment {
+  readonly manage: ClientAPI
+  readonly space: Promise<Space>
+  readonly environments: Promise<Collection<Environment, EnvironmentProps>>
+  constructor(readonly options: ContentfulOptions) {
+    this.manage = this.createClient()
+    this.space = this.getSpace()
+    this.environments = this.getEnvironments()
+  }
+
+  private createClient() {
+    return createClient({
+      accessToken: this.options.accessToken,
+      timeout: this.options.timeoutMs,
+    })
+  }
+
+  private async getSpace(): Promise<Space> {
+    return await this.manage.getSpace(this.options.spaceId)
+  }
+
+  public async getEnvironments(): Promise<
+    Collection<Environment, EnvironmentProps>
+  > {
+    return await (await this.space).getEnvironments()
+  }
+
+  public async getEnvironment(environmentId: string): Promise<Environment> {
+    return await (await this.space).getEnvironment(environmentId)
+  }
+
+  public async deleteEnvironment(environmentId: string): Promise<void> {
+    const environment = await (await this.space).getEnvironment(environmentId)
+    return environment.delete()
+  }
+
+  public async createEnvironmentWithId(
+    environmentId: string
+  ): Promise<Environment> {
+    return (await this.space).createEnvironmentWithId(environmentId, {
+      name: environmentId,
```

Comment:
should the name be the same as the `environmentId` or it could be another name?

Reply:
usually it will be the same